### DISCUSSION TITLE
change seed in cp solver c++?

### DISCUSSION THREAD

**Discussion by 515orestis** [google/or-tools#2458] [Routing (and legacy CP) questions] [answered]
i am using  cp solver to solve a routing problem i am trying unsuccessfully to introduce  randomness
on my problem( using reseed) is there a flag or any other specific configuration that i need to do?

using or tools 7.1
and best_insertions

if you need more info i will be more than happy to help

**Comment by Mizux** [comment]
in C++ (on master branch!) we have:
https://github.com/google/or-tools/blob/2fc5d0c6810600f5a1664b6b8dccb7b728b15d16/ortools/constraint_solver/constraint_solver.cc#L89-L92

https://github.com/google/or-tools/blob/2fc5d0c6810600f5a1664b6b8dccb7b728b15d16/ortools/constraint_solver/constraint_solver.h#L92-L94
which won't be binded in Python, Java or .Net

**Comment by vgala** [comment]
How can i achieve the same in Python ?

**Comment by PeterSR** [comment]
@Mizux - Is this still not possible to control via Python?

And does it mean that if we don't change it, it will be deterministic?

**Comment by lperron** [comment, marked as answer]
Just parse the header file:

https://github.com/google/or-tools/blob/5425dedcfbb22cb74c636c1374a9b5ad684b1eb5/ortools/constraint_solver/constraint_solver.h#L2929

Which is correctly wrapped in python.
